recipes/speedb/all/conanfile.py

```diff
@@ -0,0 +1,138 @@
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_snappy": True,
+        "with_lz4": True,
+        "with_zlib": True,
+        "with_zstd": True,
+        "with_core_tools": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12",
+            "Visual Studio": "16",
+            "msvc": "192",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        # TODO: support jemalloc, liburing
+        if self.options.with_snappy:
+            self.requires("snappy/1.1.10")
+        if self.options.with_lz4:
+            self.requires("lz4/1.9.4")
+        if self.options.with_zlib:
+            self.requires("zlib/[>=1.2.11 <2]")
+        if self.options.with_zstd:
+            self.requires("zstd/1.5.5")
+        if self.options.with_core_tools:
+            self.requires("gflags/2.2.2")
+            self.requires("readline/8.2")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["WITH_JEMALLOC"] = False
+        tc.variables["WITH_LIBURING"] = False
+        tc.variables["WITH_SNAPPY"] = self.options.with_snappy
+        tc.variables["WITH_LZ4"] = self.options.with_lz4
+        tc.variables["WITH_ZLIB"] = self.options.with_zlib
+        tc.variables["WITH_ZSTD"] = self.options.with_zstd
+        tc.variables["WITH_TESTS"] = False
+        tc.variables["WITH_BENCHMARK_TOOLS"] = False
+        tc.variables["WITH_CORE_TOOLS"] = self.options.with_core_tools
+        tc.variables["WITH_GFLAGS"] = self.options.with_core_tools
+        tc.variables["WITH_TOOLS"] = False
+        tc.variables["ROCKSDB_BUILD_SHARED"] = self.options.shared
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        # to avoid wrong CPU detection, modify not to use "-march=native" in apple-clang
```

Comment:
@toge Is it to be able to compiler in Mac M1? If yes, could you please report the upstream about the case?